batch.go

```diff
@@ -7,8 +7,16 @@ package ach
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
+	"math/rand"
 	"strconv"
 	"strings"
+	"time"
+	"unicode/utf8"
+)
+
+var (
+	traceNumberSource = rand.NewSource(time.Now().Unix())
```

Comment:
What is traceNumberSource used for?

Reply:
We randomly generate a TraceNumber for the offset record. Maybe we should inherit from the surrounding entries? I didn't want to get too far into parsing TraceNumbers and incrementing as we might collide with a 3rd party incrementing scheme.

Reply:
I would think the sequence number needs to be increment from what is previously in the Batch? Should we check in batch.build if this is a balanced file and increment it in this section? 
https://github.com/moov-io/ach/blob/21375f23ca5ce5ef72629a3fbb567298cafdccd0/batch.go#L362

Reply:
Yea. The TraceNumber now comes from previous EntryDetail records.